### PR TITLE
Removed Plumbum dependency

### PR DESCRIPTION
**Pull request by zaidoon1**
The following changes have been made:
1) Removed all Plumbum dependency as Plumbum has important bugs that have not been fixed since 2017
2) All string inputs  to functions are now being "shell-escaped" so that they can safely be used as arguments in a shell command line

### CODE REVIEW

File: src/data/data_controller.py [hunk 1]
```diff
@@ -1,34 +1,40 @@
...
     def __init__(self, blockchain_name: str):
-        self._multichain = local['multichain-cli'][blockchain_name]
-        self._publish_item = self._multichain['publish']
-        self._get_stream_key_items = self._multichain['liststreamkeyitems']
-        self._get_stream_keys_items = self._multichain['liststreamqueryitems']
-        self._get_stream_keys = self._multichain['liststreamkeys']
-        self._get_stream_items = self._multichain['liststreamitems']
-        self._get_stream_publisher_items = self._multichain['liststreampublisheritems']
-        self._get_stream_publishers = self._multichain['liststreampublishers']
+        self._multichain_arg = ['multichain-cli', quote(blockchain_name)]
```
Comment (yeskunall) [inline src/data/data_controller.py:10]: This makes me happy.

File: src/data/data_controller.py [hunk 2]
```diff
@@ -1,34 +1,40 @@
...
-            output = self._publish_item[
-                stream, json.dumps(keys), formatted_data].run(retcode=0)
+            args = self._publish_item_arg + \
+                [quote(stream), json.dumps(keys), formatted_data]
+            output = run(args, check=True, capture_output=True)
 
-            return output[1].strip()
-        except ProcessExecutionError as err:
-            print(err.args[3])
+            return output.stdout.strip()
```
Comment (yeskunall) [inline src/data/data_controller.py:35]: This makes me so happy.

File: .gitignore [hunk 3]
```diff
@@ -7,3 +7,9 @@ src/encryption/__pycache__/asymmetric_key_pair.cpython-36.pyc
 src/encryption/__pycache__/encryption_controller.cpython-36.pyc
 src/encryption/__pycache__/key_data.cpython-36.pyc
 src/encryption/__pycache__/encoded_asymmetric_key_pair.cpython-36.pyc
+src/data/__pycache__/data_controller.cpython-37.pyc
+src/data/__pycache__/data_stream_controller.cpython-37.pyc
+src/encryption/__pycache__/asymmetric_key_pair.cpython-37.pyc
+src/encryption/__pycache__/encoded_asymmetric_key_pair.cpython-37.pyc
+src/encryption/__pycache__/encryption_controller.cpython-37.pyc
+src/encryption/__pycache__/key_data.cpython-37.pyc
```
Comment (yeskunall) [inline .gitignore]: We can’t be adding compiled source(s) for every new file we import. Can you incorporate the `.gitignore` file like I asked you to in #2? Here’s a [sample](https://www.gitignore.io/api/python). Just do this in your terminal:

```shell
$ curl -L -s https://www.gitignore.io/api/python > .gitignore
```

... and we’re good to go (for the foreseeable future)

**Review by yeskunall — CHANGES_REQUESTED** [review]
I’ll approve once you add the `.gitignore` stuff. Good work, @zaidoon1. I have set the estimate to 5, but in the future, please label in-progress PRs as so. Currently we don’t have too many PRs, so they are easy to find. Just thinking ahead.

File: .gitignore (hunk 3 above)
Comment (zaidoon1) [reply .gitignore]: Changes have been made.

**Review by yeskunall — APPROVED** [review]

**Review by darshgrewal — APPROVED** [review]

**Review by DamanjitBhangoo — APPROVED** [review]

**Pull request merged**